### PR TITLE
Apply LoadCRD into chartCRD resource and integration tests

### PR DESCRIPTION
**Pull request by tomahawk28**
Toward giantswarm/roadmap#219

This PR is a pre-step before updating app-operator to use catalog CRD.

## Checklist

- [x] Update changelog in CHANGELOG.md.

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -5,8 +5,9 @@ go 1.16
 require (
 	github.com/Masterminds/semver/v3 v3.1.1
 	github.com/ghodss/yaml v1.0.0
-	github.com/giantswarm/apiextensions/v3 v3.22.0
+	github.com/giantswarm/apiextensions/v3 v3.26.0
 	github.com/giantswarm/app/v4 v4.13.0
+	github.com/giantswarm/app/v5 v5.0.0-20210603090121-ada93de65e7f
```
Comment (tomahawk28) [inline go.mod:10]: TODO; update this to 5.0.0 before merge

Comment (tomahawk28) [reply go.mod:10]: I will not release a new app-operator version until we get rid github.com/giantswarm/app/v4

File: service/internal/crdcache/cache.go [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+
+	"github.com/giantswarm/app/v5/pkg/crd"
+	"github.com/giantswarm/microerror"
+	"github.com/giantswarm/micrologger"
+	gocache "github.com/patrickmn/go-cache"
+	apiextensionsv1 "k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/v1"
+)
+
+const (
+	expiration = 4 * time.Hour
```
Comment (tomahawk28) [inline service/internal/crdcache/cache.go:16]: This resource would fetch the crd once every 4 hours. 🚀

File: integration/setup/setup.go [hunk 3]
```diff
@@ -70,7 +69,12 @@ func installResources(ctx context.Context, config Config) error {
 		for _, crdName := range crds {
 			config.Logger.Debugf(ctx, "ensuring %#q CRD exists", crdName)
 
-			err := config.K8sClients.CRDClient().EnsureCreated(ctx, crd.LoadV1("application.giantswarm.io", crdName), backoff.NewMaxRetries(7, 1*time.Second))
+			crd, err := config.CRDGetter.LoadCRD(ctx, "application.giantswarm.io", crdName)
```
Comment (tomahawk28) [inline integration/setup/setup.go:72]: As you see, it supports almost the same convention we used it from apiextensions.

File: service/internal/crdcache/cache.go (hunk 2 above)
Comment (rossf7) [reply service/internal/crdcache/cache.go:16]: Thanks that TTL makes sense to me. If we need to refresh it sooner can we just restart the operator?

**Review by rossf7 — APPROVED** [review]

File: service/internal/crdcache/cache.go (hunk 2 above)
Comment (tomahawk28) [reply service/internal/crdcache/cache.go:16]: @rossf7 Yes, restarting the pod will load CRDs from GitHub again.

**Pull request merged**